### ISSUE TITLE
MapStore does not work for some reason

### ISSUE THREAD

**Issue by dsukhoroslov** [hazelcast/hazelcast#7214]
HZ 3.5.4.
I have a number of MapStore beans configured for my maps. It used to work perfectly till some HZ update (not sure which one). Now the stores only load data via loadAllKeys/load/loadAll methods, but store/storeAll methods are never invoked any more. I do change data in cache and see the changes, but they're not propagated to the underlying store. 
My MapStotres are configureted as write-behind, but change of it to write-though did not help. Does anyone know what could cause such issue?

Thanks, Denis

**Comment by ahmetmircik** [comment]
Hi @dsukhoroslov , 
Do you have a reproducer test? Do you see any exception in logs? What was the working HZ version?

**Comment by dsukhoroslov** [comment]
Hi @ahmetmircik 
No, there are no any errors/exceptions. I don't remember which version it was. It should be working at the beginning of this year, what was the current HZ version these days? I haven't tested this behaviour for a while.
My hope was that someone could point me to some config param which could cause such behaviour. Ok, I'll try to provide a test next year. 
Happy New Year!

**Comment by dsukhoroslov** [comment]
Just tested this behaviour on HZ 3.6-EA3. The issue has gone, all changes are propagated properly to underlying stores. I do not have any code/config changes between these two versions. 
Would be nice to find and fix it for HZ 3.5.4

**Comment by ahmetmircik** [comment]
Can you also try with latest snapshot?

To use the snapshot with maven, just replace ${hazelcast.version} below with `3.6-SNAPSHOT`

```<dependency>
    <groupId>com.hazelcast</groupId>
    <artifactId>hazelcast</artifactId>
    <version>${hazelcast.version}</version>
</dependency>
<repository>
    <id>sonatype-snapshots</id>
    <name>Sonatype Snapshot Repository</name>
    <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    <releases>
        <enabled>false</enabled>
    </releases>
    <snapshots>
        <enabled>true</enabled>
    </snapshots>
</repository>```

**Comment by dsukhoroslov** [comment]
Just tested it wih HZ 3.6-SNAPSHOT (hazelcast-all-3.6-20160115.145715-727.jar). Works fine, no issues at all. Please fix it for HZ 3.5.x.

Thanks, Denis

**Comment by gurbuzali** [comment]
closing this one, looks like the issue is already solved for the newer versions
